symfony/plugins/orangehrmLeavePlugin/Api/LeaveEntitlementAPI.php

```diff
@@ -0,0 +1,281 @@
+ * version 2 of the License, or (at your option) any later version.
+ *
+ * OrangeHRM is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program;
+ * if not, write to the Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA  02110-1301, USA
+ */
+
+namespace OrangeHRM\Leave\Api;
+
+use OrangeHRM\Core\Api\CommonParams;
+use OrangeHRM\Core\Api\V2\CrudEndpoint;
+use OrangeHRM\Core\Api\V2\Endpoint;
+use OrangeHRM\Core\Api\V2\EndpointResourceResult;
+use OrangeHRM\Core\Api\V2\EndpointResult;
+use OrangeHRM\Core\Api\V2\RequestParams;
+use OrangeHRM\Core\Api\V2\Validator\ParamRule;
+use OrangeHRM\Core\Api\V2\Validator\ParamRuleCollection;
+use OrangeHRM\Core\Api\V2\Validator\Rule;
+use OrangeHRM\Core\Api\V2\Validator\Rules;
+use OrangeHRM\Core\Traits\UserRoleManagerTrait;
+use OrangeHRM\Entity\Employee;
+use OrangeHRM\Entity\LeaveEntitlement;
+use OrangeHRM\Leave\Api\Model\LeaveEntitlementModel;
+use OrangeHRM\Leave\Api\ValidationRules\LeaveTypeIdRule;
+use OrangeHRM\Leave\Traits\Service\LeaveEntitlementServiceTrait;
+
+class LeaveEntitlementAPI extends Endpoint implements CrudEndpoint
+{
+    use UserRoleManagerTrait;
+    use LeaveEntitlementServiceTrait;
+
+    public const PARAMETER_BULK_ASSIGN = 'bulkAssign';
+    public const PARAMETER_LEAVE_TYPE_ID = 'leaveTypeId';
+    public const PARAMETER_FROM_DATE = 'fromDate';
+    public const PARAMETER_TO_DATE = 'toDate';
+    public const PARAMETER_ENTITLEMENT = 'entitlement';
+    public const PARAMETER_LOCATION_ID = 'locationId';
+    public const PARAMETER_SUBUNIT_ID = 'subunitId';
+
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $leaveEntitlement = $this->getLeaveEntitlementService()
+            ->getLeaveEntitlementDao()
+            ->getLeaveEntitlement($this->getIdUrlAttribute());
+        $this->throwRecordNotFoundExceptionIfNotExist($leaveEntitlement, LeaveEntitlement::class);
+        $this->checkLeaveEntitlementAccessible($leaveEntitlement);
+        return new EndpointResourceResult(LeaveEntitlementModel::class, $leaveEntitlement);
+    }
+
+    /**
+     * @return int
+     */
+    private function getIdUrlAttribute(): int
+    {
+        return $this->getRequestParams()->getInt(
+            RequestParams::PARAM_TYPE_ATTRIBUTE,
+            CommonParams::PARAMETER_ID
+        );
+    }
+
+    /**
+     * @param LeaveEntitlement $leaveEntitlement
+     */
+    private function checkLeaveEntitlementAccessible(LeaveEntitlement $leaveEntitlement): void
+    {
+        $empNumber = $leaveEntitlement->getEmployee()->getEmpNumber();
+        if (!($this->getUserRoleManager()->isEntityAccessible(Employee::class, $empNumber) ||
```

Comment:
If the entity is not accessible why it throws a record-not-found exception?

Reply:
yeah. we can use `throw $this->getForbiddenException();` even. Which approach you preferred. I have seen not found also used instead of unauthorized. Because throwing forbidden, end-user can get an idea, the entity exists.

Reply:
I prefer $this->getForbiddenException(); we can handle the prod error messages later based on `env`

Reply:
done